schism/compiler.ss

```diff
@@ -497,15 +501,17 @@
   (define (parse-body body env)
     (unless (pair? body) (error 'parse-body "Empty body"))
     (parse-begin (parse-expr (car body) env) (cdr body) env))
-  (define (parse-function function env)
-    (let* ((name (definition-name function))
-           (args (function-formals function))
-           (args* (map rename-var args))
-           (body (parse-body (function-body function)
-                             (add-lexicals args args* env))))
-      `(,(cons name args*) ,body)))
-  (define (parse-functions definitions env)
-    (map (lambda (fn) (parse-function fn env)) definitions))
+  (define (parse-definition def env)
+    (let ((name (definition-name def)))
+      (if (function-definition? def)
+          (let* ((args (function-formals def))
+                (args* (map rename-var args))
+                (body (parse-body (function-body def)
+                                  (add-lexicals args args* env))))
+            `(,(cons name args*) ,body))
+          `(define ,name ,(parse-expr (caddr def) env)))))
```

Comment:
To make it a little clearer which kind of define we're using, I'd consider parsing these into `define-function` and `define-value`. It's up to you though.

Reply:
Yeah, that sounds better, will do!